app/controllers/reports_controller.rb

```diff
@@ -24,9 +21,15 @@ def show
   def destroy
     @report = resource_base.find(params[:id])
     if @report.destroy
-      process_success :success_msg => _("Successfully deleted report."), :success_redirect => reports_path
+      process_success(:success_msg => _("Successfully deleted report."), :success_redirect => reports_path)
     else
       process_error
     end
   end
+
+  private
+
+  def check_hosts_permission
+    @resource_base = resource_base.my_reports
```

Comment:
resource_base is a method, shouldn't this override it rather than set an instance variable?  (And it should be in all controllers.)